frontend/src/metabase/visualizations/lib/scalar_utils.unit.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { TYPE } from "metabase-lib/types/constants";
+
+import {
+  compactifyValue,
+  COMPACT_WIDTH_PER_DIGIT,
+  COMPACT_MAX_WIDTH,
+  COMPACT_MIN_LENGTH,
+} from "./scalar_utils";
+
+describe("scalar utils", () => {
+  describe("compactifyValue", () => {
+    const formatOptions = {
+      column: {
+        base_type: TYPE.Number,
+        semantic_type: TYPE.Number,
+      },
+    };
+
+    it("displayValue is fullScalarValue when fullScalarValue.length <= COMPACT_MIN_LENGTH", () => {
+      const value = 45000;
+      const width = 200;
+
+      const { displayValue, fullScalarValue } = compactifyValue(
+        formatOptions,
+        value,
+        width,
+      );
+
+      if (typeof fullScalarValue !== "string") {
+        throw Error("fullScalarValue is not a string");
+      }
+      expect(fullScalarValue.length).toBeLessThanOrEqual(COMPACT_MIN_LENGTH);
+
+      expect(displayValue).toBe(fullScalarValue);
+      expect(fullScalarValue).toBe("45,000");
+    });
+
+    it("displayValue is compact when fullScalarValue.length > COMPACT_MIN_LENGTH and width < COMPACT_MAX_WIDTH", () => {
+      const value = 45000.1343;
+      const width = 200;
+
+      const { displayValue, fullScalarValue } = compactifyValue(
+        formatOptions,
+        value,
+        width,
+      );
+
+      if (typeof fullScalarValue !== "string") {
+        throw Error("fullScalarValue is not a string");
+      }
```

Comment:
We don't generally want to throw errors in tests since we want our assertions to be the only thing throwing on failure.

But, I see you did this to make typescript happy 😁 

Instead, you can add an explicit type cast in the test to tell typescript you're sure you're always getting a string:

```ts
      const { displayValue, fullScalarValue } = compactifyValue(
        formatOptions,
        value,
        width,
      ) as { displayValue: string; fullScalarValue: string};
```

Note: this is not generally a good practice in production code because it overrides a lot of typescript's usefulness, but it's often fine in tests when you control the inputs and outputs.